### ISSUE TITLE
Implementation of feedForwardMaskArray for VariationalAutoencoders

### ISSUE THREAD

**Issue by AndreListou** [eclipse/deeplearning4j#6463]
Hello,

I'm currently working on a time series regression problem. I'm using the well-known C-MAPSS dataset (Dataset nr. 6 from https://ti.arc.nasa.gov/tech/dash/groups/pcoe/prognostic-data-repository/).

A short explanation of the dataset:
The C-MAPSS dataset is divided into four subsets, and each subset is further divided into training and test sets of multiple multivariate time series. Each time series is from a different aircraft gas turbine engine and starts with different degrees of initial wear and manufacturing variation. All engines operate in normal condition at the start, then begin to degrade at some point during the time series. The degradation in the training sets grows in magnitude until failure, while the degradation in the test sets ends sometime prior to failure, that is the remaining useful life (RUL). That is, the last time step for each engine in the test sets provides the true RUL targets. Thus, the main objective is to predict the correct RUL value for each engine in the test sets. The four subsets vary in operating and fault conditions and the data is contaminated with sensor noise. Each subset includes 26 columns: engine number, time step, three operational sensor settings, and 21 sensor measurements.

Loading the datasets:
I'm using the CSVSequenceRecordReader to handle the time series data. Thus, each engine is defined in its own file. The input training data has the following shape: [miniBatchSize, inputSize, timeSeriesLength], where miniBatchSize is the number of engines in the mini batch, input size is the number of columns, and timeSeriesLength is the total number of time steps in the mini batch. The engines have variable time step lengths, and hence, the shorter engines in a mini batch are padded with zeros such that the time step lengths are equal to the longest among them. Accordingly, mask arrays are needed during training.

Deep architecture and training:
For this problem, I want to use a VariationalAutoencoder in the first layer, a LSTM in the second layer, and a RnnOutputLayer in the last layer:

![image](https://user-images.githubusercontent.com/43406985/45750095-1b657c80-bc0e-11e8-94c1-84834bd68555.png)

The VAE will be used as an initial unsupervised pre-training stage in order to initialize the weights based on the underlying degradation trends in the data:

![image](https://user-images.githubusercontent.com/43406985/45750143-4354e000-bc0e-11e8-91f5-d5d227366ac8.png)

The unsupervised pre-training stage works fine, but when model.fit(trainData) is called in the fine-tuning procedure, the following exception occurs:

![image](https://user-images.githubusercontent.com/43406985/45750313-cf670780-bc0e-11e8-9f9b-7c194c323f9f.png)

Will you implement feedForwardMaskArray for VariationalAutoencoders in the near future?

Best,

Andre Ellefsen

Aha! Link: https://skymindai.aha.io/features/DL4J-62

**Comment by AndreListou** [comment]
@AlexDBlack any news regarding this issue?

**Comment by agibsonccc** [comment]
Closing — this issue is over 7 years old with no recent activity. The codebase has undergone a major rewrite. If this is still relevant, please open a new issue with current reproduction steps.
